File: java/src/test/java/org/npb/agview/api/example/PremisePostHandlerTest.java
package org.npb.agview.api.example;

import com.fasterxml.jackson.core.JsonProcessingException;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import java.net.http.HttpClient;

import static org.hamcrest.MatcherAssert.assertThat;
import static org.hamcrest.Matchers.*;
import static org.npb.agview.api.example.Constants.*;

class PremisePostHandlerTest {

    private PremisePostHandler sut;

    private final String BASE_URL = System.getenv("NPB_BASE_URL");
    private final String API_KEY = System.getenv("NPB_API_KEY");
    private final String API_SECRET = System.getenv("NPB_API_SECRET");
    private final HttpClient httpClient = HttpClient.newHttpClient();

    @BeforeEach
    public void setup() {
        var dbHandler = new PremiseDbHandler(DB_DIRECTORY+"/premise.csv",
                DB_DIRECTORY+"/premise_address.csv", new DbHandler());
        var connectionInfo = new Arguments(BASE_URL, API_KEY, API_SECRET);
        var accessTokenHandler = new AccessTokenHandler(httpClient, connectionInfo);
        sut = new PremisePostHandler(httpClient,
                connectionInfo,
                accessTokenHandler,
                dbHandler);
    }

    @Test
    public void createsPremises() throws JsonProcessingException {
        var actual = sut.createPremises();

        assertThat(actual, hasSize(greaterThan(0)));
    }

}

File: c-sharp/ApiTests/PremisePostHandlerTest.cs
using System.Net.Http;
using System.Threading.Tasks;
using FluentAssertions;
using Npb.Agview.Api.Example;
using Xunit;

namespace Npb.Agview.ApiTests.Example
{
    public class PremisePostHandlerTest
    {
        private readonly PremisePostHandler _sut;

        private readonly HttpClient _httpClient = new();
        private readonly string BaseUrl = Constants.BaseUrl;
        private readonly string ApiKey = Constants.ApiKey;
        private readonly string ApiSecret = Constants.ApiSecret;

        public PremisePostHandlerTest()
        {
            var accessTokenHandler = new AccessTokenHandler(_httpClient, BaseUrl, ApiKey, ApiSecret);
            var premiseDbHandler = new PremiseDbHandler(TestConstants.PremisesFilePath, TestConstants.PremiseAddressesFilePath);

            _sut = new PremisePostHandler(_httpClient, accessTokenHandler, BaseUrl, premiseDbHandler);
        }

        [Fact]
        public async Task CreatesPremises()
        {
            var actual = await _sut.CreatePremises();

            actual.Should().HaveCountGreaterThan(0);
        }
    }
}
